modules/javafx.base/src/test/java/test/javafx/util/converter/LocalDateStringConverterTest.java

```diff
@@ -48,42 +53,100 @@
+    // Local*StringConverter tests and cause unpredictable results.
+    private enum LocalDateStringConverterVariant {
+        NO_PARAM,
+        WITH_FORMATTER_PARSER,
+        WITH_FORMAT_STYLES,
+    };
 
     @Parameterized.Parameters public static Collection implementations() {
         return Arrays.asList(new Object[][] {
-            { new LocalDateStringConverter(),
-              Locale.getDefault(Locale.Category.FORMAT), FormatStyle.SHORT,
-              VALID_DATE, null, null },
+            { LocalDateStringConverterVariant.NO_PARAM,
+              FormatStyle.SHORT, VALID_DATE },
 
-            { new LocalDateStringConverter(aFormatter, aParser),
-              Locale.getDefault(Locale.Category.FORMAT), null,
-              VALID_DATE, aFormatter, aParser },
+            { LocalDateStringConverterVariant.WITH_FORMATTER_PARSER,
+              null, VALID_DATE },
 
-            { new LocalDateStringConverter(FormatStyle.SHORT, Locale.UK, IsoChronology.INSTANCE),
-              Locale.UK, FormatStyle.SHORT,
-              VALID_DATE, null, null },
+            { LocalDateStringConverterVariant.WITH_FORMAT_STYLES,
+              FormatStyle.SHORT, VALID_DATE },
         });
     }
 
+    private LocalDateStringConverterVariant converterVariant;
+    private FormatStyle dateStyle;
+    private LocalDate validDate;
+    
     private LocalDateStringConverter converter;
     private Locale locale;
-    private FormatStyle dateStyle;
     private DateTimeFormatter formatter, parser;
-    private LocalDate validDate;
 
-    public LocalDateStringConverterTest(LocalDateStringConverter converter, Locale locale, FormatStyle dateStyle, LocalDate validDate, DateTimeFormatter formatter, DateTimeFormatter parser) {
-        this.converter = converter;
-        this.locale = locale;
+    public LocalDateStringConverterTest(LocalDateStringConverterVariant converterVariant, FormatStyle dateStyle, LocalDate validDate) {
+        this.converterVariant = converterVariant;
         this.dateStyle = dateStyle;
         this.validDate = validDate;
-        this.formatter = formatter;
-        this.parser = parser;
+
+        // initialized after Locale is established
+        this.converter = null;
+        this.locale = null;
+        this.formatter = null;
+        this.parser = null;
+    }
+
+    @BeforeClass
+    public static void setupBeforeAll() {
+        // Tests require that default locale is en_US
+        oldLocale = Locale.getDefault();
+        Locale.setDefault(Locale.US);
+
+        // DateTimeFormatter uses default locale, so we can init this after updating locale
+        aFormatter = DateTimeFormatter.ofPattern("dd MM yyyy");
+        aParser = DateTimeFormatter.ofPattern("yyyy MM dd");
+    }
+
+    @AfterClass
+    public static void teardownAfterAll() {
+        // Restore VM's old locale
+        Locale.setDefault(oldLocale);
+    }
+
+    @Before
+    public void setup() {
+        if (this.converter == null) {
```

Comment:
There is no need to check for `null` here. It always will be, since `converter` is initialized to `null` in the constructor and there is no other assignment.

Reply:
Done, also fixed in other files